test/partition-healing-tests.js

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+var TestCoordinator = require('./test-coordinator');
+var _ = require('lodash');
+var dsl = require('./ringpop-assert');
+var events = require('./events');
+var prepareCluster = require('./test-util').prepareCluster;
+var safeJSONParse = require('./util').safeParse;
+var test2 = require('./test-util').test2;
+
+// First stage of healing algorithm: forced reincarnation.
+//
+// Actors:
+// ● a1, sut: partition A.
+// ● b1, b2: partition B.
+//
+// 1. Create two partitions A and B by marking b1 and b2 failed in sut.
+//    a1 and sut are failed according to B.
+// 2. Trigger partition healing.
+// 3. Verify sut tries to reincarnate both sides:
+// 3.1. sut declares b1 and b2 as suspects to B.
+// 3.2. sut declares a1 as suspect to A.
+test2('reincarnating partitions A and B', [3], 20000,
```

Comment:
I implemented `healerUnknownNodes` with arbitrary number of fake nodes (with minimum being 2). That was a pretty simple test, and it worked perfectly. While I agree with your comment that parameterizing the cluster size would match the style of the rest of the tests, I don't think the value is worthwhile for added complexity.

The biggest problem in the other two tests is splitting and maintaining the membership information of fake nodes. It is not trivial with 3 nodes already, so it's going to be even more difficult with `n`.

Do you think there is another argument besides being consistent with the rest of the code base? If not, I would rather not add even more complexity to the two tests which have this non-parameterized.